setup.py

```diff
@@ -822,6 +868,58 @@ def prefix(directory, files):
     'cflags': ['-DBIT_STREAM_WORD_TYPE=uint8'],
     })
 
+# SZ library and its hdf5 filter
+sz_dir = os.path.join("src", "SZ", "sz")
+sz_sources = glob(os.path.join(sz_dir, "src", "*.c"))
+sz_include_dirs = [os.path.join(sz_dir, "include"), sz_dir]
+sz_sources += glob('src/SZ/zstd/*/*.c')
+# TODO sz_depends += glob('src/SZ/zstd*/*/*.h')
+
+ZLIB_FROM_BLOSC = True
+if ZLIB_FROM_BLOSC:
+    sz_sources += glob(blosc_dir + 'internal-complibs/zlib*/*.c')
+    ##depends += glob(blosc_dir + 'internal-complibs/zlib*/*.h')
+else:
+    sz_sources += glob('src/SZ/zlib/*.c')
+
+sz_include_dirs += glob('src/SZ/zstd')
+sz_include_dirs += glob('src/SZ/zstd/common')
+
+if ZLIB_FROM_BLOSC:
+    sz_include_dirs += glob(blosc_dir + 'internal-complibs/zlib*')
+else:
+    sz_include_dirs += glob('src/SZ/zlib')
+
+sz_include_dirs += glob('src/SZ/')
+sz_lib = ("sz", {
+    "sources": sz_sources,
+    "include_dirs": sz_include_dirs,
+    #"cflags": ["-lzstd"],
+})
+
+# "cflags": ['-DBUILD_HDF5_FILTER:BOOL=ON'],
+
+
+h5zsz_dir = os.path.join("src", "SZ", "hdf5-filter", "H5Z-SZ")
+sources = glob(h5zsz_dir + "/src/" + "*.c")
+sources += sz_sources
+depends = glob(h5zsz_dir + "/include/" + "*.h")
+include_dirs = [os.path.join(sz_dir, 'include'), os.path.join(h5zsz_dir, 'include')]
+include_dirs += sz_include_dirs
+extra_compile_args = ['-O3', '-ffast-math', '-std=c99', '-fopenmp']
+extra_compile_args += ['/Ox', '/fp:fast', '/openmp']
+extra_link_args = ['-fopenmp', '/openmp', "-lm"]
+
+sz_plugin = HDF5PluginExtension(
+    "hdf5plugin.plugins.libh5sz",
+    sources=sources,
+    depends=depends,
+    include_dirs=include_dirs,
+    extra_compile_args=extra_compile_args,
+    extra_link_args=extra_link_args,
+)
+PLUGIN_LIB_DEPENDENCIES['sz'] = 'sz'
```

Comment:
Not sure this is needed as `sz_sources` is added to `sources`, but we can check that in another PR.

Reply:
OK